storage/src/rocksdb_client.cpp

```diff
@@ -81,35 +92,24 @@ bool Client::isNew() {
  * Uses the RocksDBClient object variables m_dbPath and m_dbInstance to
  * establish a connection with RocksDB by creating a RocksDB object.
  *
- *  @return GeneralError in case of error in connection, else OK.
+ *  @throw GeneralError in case of error in connection, else OK.
  */
-Status Client::init(bool readOnly) {
-  ::rocksdb::DB *db;
+void Client::init(bool readOnly) {
+  ::rocksdb::DB* db;
   ::rocksdb::Options options;
+  ::rocksdb::TransactionDBOptions txn_options;
   options.create_if_missing = true;
   options.comparator = m_comparator;
-
   ::rocksdb::Status s;
   if (readOnly) {
     s = ::rocksdb::DB::OpenForReadOnly(options, m_dbPath, &db);
+    m_dbInstance.reset(db);
   } else {
-    s = ::rocksdb::DB::Open(options, m_dbPath, &db);
+    s = ::rocksdb::TransactionDB::Open(options, txn_options, m_dbPath, &txn_db_);
```

Comment:
This changes general RocksDB behavior. Before the product will start working using bft-engine storage, verification tests should be launched.

Reply:
All the actions except beginTransaction are performed using m_dbInstance(which is rocksdb::DB*) as previously
Verification tests are always good